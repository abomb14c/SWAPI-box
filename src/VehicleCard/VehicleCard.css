*,
*:after,
*:before {
box-sizing: border-box;
margin: 0;
}

hr {
    text-align: center;
    margin:0 auto;
    width: 60%;
}

.description-title{
    color: #EB544D;
    margin-bottom: 1%;
}

.description-content {
    font-size: 24px;
}

.content-div {
    margin: 15px;
}

.vehicle-card {
    background-color: #ffffff;
    border: 2px solid #D6D6D6;
    display: inline-block;
    height: 400px;
    left: 15px;
    margin: 3px;
    position: relative;
    vertical-align: top;
    width: 340px;
}

.vehicle-name {
    color: rgba(255,255,255);
    margin: none;
    text-align: center;
    text-shadow: 4px 4px #000;
}

.vehicle-name-section {
    background-image: url('../assets/vehicles-section-header.jpg');
    background-size: cover;
    font-size: 36px;
    height: 140px;
    padding-top: 50px;
}



.residents{
    font-size: 18px;
}
