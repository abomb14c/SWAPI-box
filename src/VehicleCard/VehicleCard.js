import React from 'react';
import './VehicleCard.css';
import PropTypes from 'prop-types';

const VehicleCard = (props) => {
  const{name, model, type, passengers, findFavoriteVehicle, id, category} = props;

  return (
    <article className="vehicle-card">
      <section className="vehicle-name-section">
        <h1 className="vehicle-name">{name}</h1>
        <button onClick={()=>findFavoriteVehicle(id)} className="favorites">Favorite</button>
      </section>
      <div className="content-div">
        <p className="description-title">Model</p>
        <h3 className="description-content">{model}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Class</p>
        <h3 className="description-content">{type}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Passengers</p>
        <h3 className="description-content">{passengers}</h3>
      </div>
    </article>
  );
    
};
VehicleCard.propTypes = {
  name: PropTypes.string,
  model: PropTypes.string,
  type: PropTypes.string,
  passengers: PropTypes.number,
  findFavoriteVehicle: PropTypes.func,
  id: PropTypes.string,
  category: PropTypes.string
};

export default VehicleCard;



