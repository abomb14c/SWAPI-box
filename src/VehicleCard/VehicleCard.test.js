import React from 'react'
import ReactDOM from 'react-dom';
import VehicleCard from './VehicleCard'
import {shallow, mount} from 'enzyme';

it('should match the snapshot', () => {
    let name='Roxanne'
    let model='subaru'
    let type= "bitchin"
    let findFavorite= jest.fn()
    let id = "Vehicle0"
    let category="vehicles"

    let wrapper= shallow(<VehicleCard name={name}
    model={model}
    type={type}
    findFavoriteVehicle={findFavorite}
    id={id}
    category={category}
    />)
    expect(wrapper).toMatchSnapshot();
  })