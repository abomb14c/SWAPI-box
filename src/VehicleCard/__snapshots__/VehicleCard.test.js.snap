// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VehicleCard
    category="vehicles"
    findFavoriteVehicle={[Function]}
    id="Vehicle0"
    model="subaru"
    name="Roxanne"
    type="bitchin"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="vehicle-name-section"
>
          <h1
                    className="vehicle-name"
          >
                    Roxanne
          </h1>
          <button
                    className="favorites"
                    onClick={[Function]}
          >
                    Favorite
          </button>
</section>,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Model
          </p>
          <h3
                    className="description-content"
          >
                    subaru
          </h3>
</div>,
        <hr />,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Class
          </p>
          <h3
                    className="description-content"
          >
                    bitchin
          </h3>
</div>,
        <hr />,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Passengers
          </p>
          <h3
                    className="description-content"
          />
</div>,
      ],
      "className": "vehicle-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="vehicle-name"
>
              Roxanne
</h1>,
            <button
              className="favorites"
              onClick={[Function]}
>
              Favorite
</button>,
          ],
          "className": "vehicle-name-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Roxanne",
              "className": "vehicle-name",
            },
            "ref": null,
            "rendered": "Roxanne",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "className": "favorites",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Model
</p>,
            <h3
              className="description-content"
>
              subaru
</h3>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Model",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Model",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "subaru",
              "className": "description-content",
            },
            "ref": null,
            "rendered": "subaru",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Class
</p>,
            <h3
              className="description-content"
>
              bitchin
</h3>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Class",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Class",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "bitchin",
              "className": "description-content",
            },
            "ref": null,
            "rendered": "bitchin",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Passengers
</p>,
            <h3
              className="description-content"
/>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Passengers",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Passengers",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "description-content",
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="vehicle-name-section"
>
            <h1
                        className="vehicle-name"
            >
                        Roxanne
            </h1>
            <button
                        className="favorites"
                        onClick={[Function]}
            >
                        Favorite
            </button>
</section>,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Model
            </p>
            <h3
                        className="description-content"
            >
                        subaru
            </h3>
</div>,
          <hr />,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Class
            </p>
            <h3
                        className="description-content"
            >
                        bitchin
            </h3>
</div>,
          <hr />,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Passengers
            </p>
            <h3
                        className="description-content"
            />
</div>,
        ],
        "className": "vehicle-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="vehicle-name"
>
                Roxanne
</h1>,
              <button
                className="favorites"
                onClick={[Function]}
>
                Favorite
</button>,
            ],
            "className": "vehicle-name-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Roxanne",
                "className": "vehicle-name",
              },
              "ref": null,
              "rendered": "Roxanne",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorite",
                "className": "favorites",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Favorite",
              "type": "button",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Model
</p>,
              <h3
                className="description-content"
>
                subaru
</h3>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Model",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Model",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "subaru",
                "className": "description-content",
              },
              "ref": null,
              "rendered": "subaru",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Class
</p>,
              <h3
                className="description-content"
>
                bitchin
</h3>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Class",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Class",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "bitchin",
                "className": "description-content",
              },
              "ref": null,
              "rendered": "bitchin",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Passengers
</p>,
              <h3
                className="description-content"
/>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Passengers",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Passengers",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "description-content",
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
